### PR TITLE
Show application icon when using sway window module

### PR DESCRIPTION
**Pull request by FlexW**
Demo:

https://user-images.githubusercontent.com/19961516/137642147-1f71a58e-a91c-41d0-8124-e40e895153e0.mp4

Signed-off-by: Felix Weilbach <felix.weilbach@t-online.de>

### CODE REVIEW

**Comment by FlexW** [comment]
@Alexays Thanks for this nice project. Could you please give me some feedback if that is a feature that might get accepted?

File: src/modules/sway/window.cpp [hunk 1]
```diff
@@ -29,12 +38,60 @@ void Window::onCmd(const struct Ipc::ipc_response& res) {
...
+  for (const auto& data_dir : data_dirs) {
+    const auto desktop_file_path = data_dir + "applications/" + app_id + ".desktop";
+    if (std::filesystem::exists(desktop_file_path)) {
+      return desktop_file_path;
+    }
+  }
+  return {};
+}
+
+std::optional<Glib::ustring> getIconName(const std::string& app_id) {
```
[the remark this answers is not in the document]
Comment (FlexW) [reply src/modules/sway/window.cpp:59]: Are you speaking of the code in `modules/sni/item.cpp`? The difference is, that the sni item gets its icon/icon name via dbus. The sni module does not need to scan the desktop file as I do with the function `getIconName()`. My conclusion would be, that we can't reuse any code here.

[the remark this answers is not in the document]
Comment (Alexays) [reply src/modules/sway/window.cpp:59]: Sorry for the delay, ok my bad

**Comment by Alexays** [comment]
LGTM, can you just update man? :)

File: src/modules/sway/window.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (FlexW) [reply src/modules/sway/window.cpp:59]: no problem :)

**Comment by FlexW** [comment]
> LGTM, can you just update man? :)

Updated;)

**Comment by rieje** [comment]
This is not documented in man? With `"height": 20,` in ~/.config/waybar/config, icon is not scaled down to fit this so the bar height changes between 20 when there is no icon to something bigger, but I want 20 to be the highest height. Lastly, is there an option to disable the icon?

**Comment by WhyNotHugo** [comment]
Oh, interesting feature. Just updated waybar and suddenly the icons showed up, interesting.

I noticed that the icon for XWayland apps don't show up. Is this a known issue?

**Comment by Xiretza** [comment]
> > LGTM, can you just update man? :)
> 
> Updated;)

 Doesn't look like that actually ever happened...

**Comment by FlexW** [comment]
> > > LGTM, can you just update man? :)
> > 
> > 
> > Updated;)
> 
> Doesn't look like that actually ever happened... Edit: ah, fixed by #1459.

Actually I misunderstood it and thought that the request was just to update (rebase) the pull request. Sorry for that.

**Pull request merged**